src/main/java/de/lh/tool/service/entity/impl/ItemServiceImpl.java

```diff
@@ -1,70 +1,255 @@
+import de.lh.tool.service.entity.interfaces.TechnicalCrewService;
+import de.lh.tool.service.entity.interfaces.UserRoleService;
+import de.lh.tool.util.DateUtil;
+import de.lh.tool.util.ValidationUtil;
 
 @Service
 public class ItemServiceImpl extends BasicMappableEntityServiceImpl<ItemRepository, Item, ItemDto, Long>
 		implements ItemService {
 
 	@Autowired
 	private ItemHistoryService itemHistoryService;
+	@Autowired
+	private UserRoleService userRoleService;
+	@Autowired
+	private ProjectService projectService;
+	@Autowired
+	private SlotService slotService;
+	@Autowired
+	private TechnicalCrewService technicalCrewService;
+	@Autowired
+	private ItemItemService itemItemService;
+
+	@Override
+	public boolean isViewAllowed(Item item) {
+		boolean ownProject = Optional.ofNullable(item).map(Item::getSlot).map(Slot::getStore)
+				.map(Store::getStoreProjects).map(storeProjects -> storeProjects.stream().anyMatch(storeProject -> {
+					boolean inRange = DateUtil.isDateWithinRange(LocalDate.now(), storeProject.getStart(),
+							storeProject.getEnd());
+					boolean permissionOnProject = projectService.isOwnProject(storeProject.getProject());
+					return inRange && permissionOnProject;
+				})).orElse(false);
+		boolean hasRight = userRoleService.hasCurrentUserRight(UserRole.RIGHT_ITEMS_GET_FOREIGN_PROJECT);
+		return ownProject || hasRight;
+	}
 
 	@Override
 	@Transactional
-	public List<ItemDto> getItemDtos() {
-		return convertToDtoList(findAll());
+	public List<ItemDto> findItemDtosByFilters(String freeText) {
+		return convertToDtoList(getRepository().findByFilters(freeText).stream().filter(this::isViewAllowed)
+				.collect(Collectors.toList()));
 	}
 
 	@Override
 	@Transactional
-	public ItemDto getItemDtoById(Long id) throws DefaultException {
-		Item item = findById(id).orElseThrow(() -> new DefaultException(ExceptionEnum.EX_INVALID_ID));
+	public ItemDto findItemDtoById(Long id) throws DefaultException {
+		Item item = findItemByIdIfAllowed(id);
 		return convertToDto(item);
 	}
 
 	@Override
 	@Transactional
 	public ItemDto createItemDto(ItemDto dto) throws DefaultException {
 		if (dto.getId() != null) {
-			throw new DefaultException(ExceptionEnum.EX_ID_PROVIDED);
+			throw ExceptionEnum.EX_ID_PROVIDED.createDefaultException();
+		}
+		Item item = getValidatedItem(dto);
+		if (!isViewAllowed(item)) {
+			throw ExceptionEnum.EX_FORBIDDEN.createDefaultException();
 		}
-		Item item = save(convertToEntity(dto));
+		item = save(item);
+		itemHistoryService.logCreated(item);
+
 		return convertToDto(item);
 	}
 
 	@Override
 	@Transactional
 	public ItemDto updateItemDto(ItemDto dto, Long id) throws DefaultException {
 		dto.setId(ObjectUtils.defaultIfNull(id, dto.getId()));
-		if (dto.getId() == null) {
-			throw new DefaultException(ExceptionEnum.EX_NO_ID_PROVIDED);
+		ValidationUtil.checkIdsNonNull(dto.getId());
+		Item itemToCompare = getValidatedItem(dto);
+
+		Item old = findItemByIdIfAllowed(dto.getId());
```

Comment:
Könnte old null sein?

Reply:
nein, dann fliegt eine Exception:
Item item = findById(id).orElseThrow(ExceptionEnum.EX_INVALID_ITEM_ID::createDefaultException);
Ich hab mal ein @NonNull an den Rückgabewert in der Methodendeklaration gehängt ;)